File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/chat/impl/ChatModule.kt
package au.com.crazybean.mobilex.kurir.modules.chat.impl

import au.com.crazybean.mobilex.kurir.modules.chat.ChatDelegate
import au.com.crazybean.mobilex.kurir.modules.chat.ChatView
import au.com.crazybean.mobilex.kurir.modules.chat.ChatViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val chatModule = module {
    // ViewModel
    viewModel {
        ChatViewModel(get(), get())
    }

    // Delegate
    factory { (view: ChatView?) ->
        ChatDelegate(view, get())
    }
}

File: ios/Kurir/modules/chat/impl/ChatModule.swift
//
//  ChatModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Crazybean

class ChatModule: Module {
    override func inject() {
        // ViewModel
        factory(ChatViewModel.self) { _ in
            ChatViewModel()
        }
        
        // Delegate
        factory(ChatDelegate.self) { (r, view: ChatView) -> ChatDelegate in
            ChatDelegate(view: view, viewModel: r.resolve(ChatViewModel.self)!)
        }
    }
}
